framework/src/seclink/seclink.c

```diff
@@ -174,6 +174,48 @@ int sl_deinit(sl_ctx hnd)
 	return 0;
 }
 
+#ifdef CONFIG_ARM_TRUSTZONE
+int sl_allocate(sl_ctx hnd, uint32_t ssize)
+{
+	SL_ENTER;
+	if (!hnd || ((struct _seclink_s_ *)hnd)->fd <= 0) {
+		SL_ERR(-1);
+		return SECLINK_ERROR;
```

Comment:
SECURITY_ALLOC_ERROR is correct? SECLINK_ERROR is defined in seclink.h, maybe f/w & driver will use it(of course some part of application, but it is better than return value which is defined in f/w)

Reply:
I understand your point, but the error is reported if either hnd is NULL or fd is invalid. Both such conditions are checked in sl_init, and an error there is reported as SECLINK_ERROR (line:130/line:137) , so I used the same as sl_init error return